metricq/drain.py

```diff
@@ -40,22 +41,22 @@
 
 
 class Drain(Sink):
-    def __init__(self, *args: Any, queue: str, metrics: List[str], **kwargs: Any):
+    def __init__(self, *args: Any, queue: str, metrics: Iterable[str], **kwargs: Any):
         """Drain the given queue of all buffered metric data
 
         Args:
             queue: The name of the queue that contains the subscribed data.
-            metrics: List of metrics that you want to subscribe to.
+            metrics: Metrics that you want to subscribe to.
         """
         super().__init__(*args, add_uuid=True, **kwargs)
-        if not metrics:
-            raise ValueError("Metrics list must not be empty")
         if not queue:
             raise ValueError("Queue must not be empty")
         self._queue = queue
-        self._metrics = metrics
+        self._metrics = list(metrics)
+        if not self._metrics:
```

Comment:
Technically speaking, this is now a different behavior, when `metrics` is passed as `None`. Instead of raising `ValueError`, `list(None)` raises a `TypeError`. Having said that, accounding to the type hint, it is not allowed to pass `None` for `metrics`

Reply:
I don't care about changing "UB".
By the way, this is also different behavior in case someone initializes a drain with an externally owned list and then changes that list. Which would have been all sorts of raining cats before and is now well-behaved.